public.yaml

```diff
@@ -120,11 +120,15 @@ metadata:
         "name": "salt-master-config",
         "image": "sles12/velum:__TAG__",
         "command": ["sh", "-c", "umask 377;
-                                 rmdir /salt-master-credentials/returner-credentials.conf;
-                                 if [ ! -f /salt-master-credentials/returner-credentials.conf ]; then
-                                   echo \"mysql:\" > /salt-master-credentials/returner-credentials.conf;
-                                   echo -n \"  pass: \" >> /salt-master-credentials/returner-credentials.conf;
-                                   cat /infra-secrets/mariadb-salt-password >> /salt-master-credentials/returner-credentials.conf;
+                                 rmdir /salt-master-credentials/55-returner-credentials.conf;
```

Comment:
doesn't `rmdir` fail when it's a file? but it has been there before, so there must be some reason?

Reply:
This is exactly the point. If the file does not exist, the kubelet will `mkdir` it (because of the `mountPath`), so if it is a directory we `rmdir` it, if it's a file it will do anything and we just continue.

Reply:
so if the file is there when the code is run, the `rmdir` statement will fail, but won't fail hard?

Reply:
in other words: `rmdir` on a file will always fail. is this a problem here?
```
% touch /tmp/somefile                                                                                                                                                                                      
% rmdir /tmp/somefile                                                                                                                                                                                            
rmdir: failed to remove '/tmp/somefile': Not a directory
% echo $?                                                                                                                                                                                                    1 ↵ ✭
1
```